recipes/pcapplusplus/cmake/conanfile.py

```diff
@@ -0,0 +1,60 @@
+import os
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake
+from conan.tools.files import copy, get, rmdir
+from conan.errors import ConanInvalidConfiguration
+
+
+required_conan_version = ">=1.57.0"
+
+class PcapplusplusConan(ConanFile):
+    name = "pcapplusplus"
+    package_type = "static-library"
+    license = "Unlicense"
+    description = "PcapPlusPlus is a multiplatform C++ library for capturing, parsing and crafting of network packets"
+    topics = ("pcap", "network", "security", "packet")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/seladb/PcapPlusPlus"
+    settings = "os", "arch", "build_type", "compiler"
+    options = {
+        "immediate_mode": [True, False],
+    }
+    default_options = {
+        "immediate_mode": False,
+    }
+
+    def requirements(self):
+        if self.settings.os == "Windows":
+            self.requires("npcap/1.70")
+        else:
+            self.requires("libpcap/1.10.1")
+
+    def validate(self):
+        if self.settings.os not in ("FreeBSD", "Linux", "Macos", "Windows"):
+            raise ConanInvalidConfiguration(f"{self.settings.os} is not supported")
+
+    def source(self):
+        get(
+            self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True
+        )
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["PCAPPP_BUILD_TESTS"] = False
+        tc.variables["PCAPPP_BUILD_EXAMPLES"] = False
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"), keep_path=False)
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["Pcap++", "Packet++", "Common++"]
```

Comment:
```suggestion
        self.cpp_info.libs = ["Pcap++", "Packet++", "Common++"]
        if self.settings.os == "Macos":
            self.cpp_info.frameworks = ["CoreFoundation", "SystemConfiguration"]
        elif self.settings.os == "Windows":
           self.settings.system_libs = ["ws2_32", "iphlpapi"]
```

I see some system libraries are required to build pcap++:

https://github.com/seladb/PcapPlusPlus/blob/v23.09/Pcap%2B%2B/CMakeLists.txt#L69

Reply:
Fixed in this commit: https://github.com/conan-io/conan-center-index/pull/20296/commits/6bc67b3f751a5c9c6830adf4ea4275de25575d72

I didn't add Linux and FreeBSD because PcapPlusPlus requires C++11 so pthreads and libm should be part of the standard library